performant-software/dm-2
# Batch image performance and UX improvements

### What this PR does

- Per #432:
  - Eliminates deadlocking by doing the move operations in batch at the end of uploads (`move_many`)
  - Sets a limit of 5 files/25 MB to reduce memory consumption
  - Makes all batch image fetch functions `async`/`await` to improve performance
  - Adds much more UX feedback for different states of upload/processing
- Unrelated: fixes a deprecation in `createBrowserHistory`

### Additional notes

Memory issues still remain to be seen on Heroku. We may need to implement a lower limit than 50 or a total size limit to reduce memory consumption.

### Status


### Steps to test

1. Same as #431 
2. Also try uploading more than 5 images at once and verify that an error message appears, and that it goes away when you retry upload with 5 or fewer
3. Same as 2 for a set of images that accumulates to more than 25 MB
